micropython/drivers/led/neopixel/neopixel.py

```diff
@@ -8,10 +8,16 @@ class NeoPixel:
     # G R B W
     ORDER = (1, 0, 2, 3)
 
-    def __init__(self, pin, n, bpp=3, timing=1):
+    def _clamp(self, v, c_min, c_max):
```

Comment:
This is definitely stylistically better, but functions are expensive in terms of code size (and therefore memory usage).

In this case, it's actually cheaper to just use `min(max(brightness), 0, 1)` directly in both places. Note also changing `0.0` to `0` etc is also helpful for size (and doesn't change the behaviour).